apps/etterlatte-oppdater-behandling/src/main/kotlin/TidshendelseService.kt

```diff
@@ -0,0 +1,156 @@
+package no.nav.etterlatte
+
+import no.nav.etterlatte.TidshendelseService.TidshendelserJobbType.AO_BP20
+import no.nav.etterlatte.TidshendelseService.TidshendelserJobbType.AO_BP21
+import no.nav.etterlatte.TidshendelseService.TidshendelserJobbType.AO_OMS67
+import no.nav.etterlatte.TidshendelseService.TidshendelserJobbType.OMS_DOED_3AAR
+import no.nav.etterlatte.TidshendelseService.TidshendelserJobbType.OMS_DOED_4MND
+import no.nav.etterlatte.TidshendelseService.TidshendelserJobbType.OMS_DOED_5AAR
+import no.nav.etterlatte.funksjonsbrytere.FeatureToggleService
+import no.nav.etterlatte.libs.common.behandling.Omregningshendelse
+import no.nav.etterlatte.libs.common.behandling.Prosesstype
+import no.nav.etterlatte.libs.common.behandling.Revurderingaarsak
+import no.nav.etterlatte.libs.common.oppgave.OppgaveType
+import no.nav.etterlatte.libs.common.oppgave.OppgaveType.AKTIVITETSPLIKT_OPPFOELGING
+import no.nav.etterlatte.libs.common.oppgave.OppgaveType.REVURDERING
+import no.nav.etterlatte.libs.common.tidspunkt.Tidspunkt
+import org.slf4j.LoggerFactory
+import java.time.LocalTime
+import java.util.UUID
+
+class TidshendelseService(
+    private val behandlingService: BehandlingService,
+    private val featureToggleService: FeatureToggleService,
+) {
+    private val logger = LoggerFactory.getLogger(TidshendelseService::class.java)
+
+    enum class TidshendelserJobbType {
+        AO_BP20,
+        AO_BP21,
+        AO_OMS67,
+        OMS_DOED_3AAR,
+        OMS_DOED_5AAR,
+        OMS_DOED_4MND,
+    }
+
+    fun haandterHendelse(hendelse: TidshendelsePacket): TidshendelseResult {
+        if (skalSkippes(hendelse)) {
+            return TidshendelseResult.Skipped
+        }
+        logger.info(
+            """Løpende ytelse: oppretter behandling/oppgave for sak ${hendelse.sakId} 
+                    behandlingsmåned=${hendelse.behandlingsmaaned}""",
+        )
+        if (skalLageOmregning(hendelse)) {
+            try {
+                return behandlingService.opprettOmregning(omregningshendelse(hendelse)).let { response ->
+                    TidshendelseResult.OpprettetOmregning(
+                        response.behandlingId,
+                        response.forrigeBehandlingId,
+                    )
+                }
+            } catch (e: Exception) {
+                logger.error("Kunne ikke opprette omregning [sak=${hendelse.sakId}]", e)
+                return opprettOppgave(hendelse)
+                    ?.let { oppgaveId -> TidshendelseResult.OpprettetOppgave(oppgaveId) }
+                    ?: throw RuntimeException("Oppgaven ble ikke opprettet!")
```

Comment:
https://nav-it.slack.com/archives/C01PG1JM9D4/p1712049776149559

Reply:
https://utvikling.intern.nav.no/teknisk/github-copilot.html

Reply:
Takk! Skriver det om slik at jeg slipper å håndtere null :)